hashicorp/waypoint
# ui: rearrange deployment routes

## Why the change?

This change adopts Ember’s conventions for master-detail views. That is, a plural parent route (`deployments`) which loads and displays the master data (the list of deployments), and a singular child route (`deployment`) which loads and displays the detail (a single deployment). Adopting this structure happens to fix #2918 as a happy side-effect.

## Any breaking changes?

This is a breaking change in two ways:

1. Paths that used to be like this:
   `/{workspace}/{project}/app/{app}/deployment/seq/{seq}`
   Are now like this:
   `/{workspace}/{project}/app/{app}/deployments/{seq}`
2. The legacy ID-based deployment routing is gone entirely

Originally I didn’t intend to make any breaking changes, but doing so dramatically simplified the codebase so overall it seemed worthwhile. We are pre-1.0 after all, and I think users will be able to recover from a 404 reasonably easily. Feel free to push back on this assertion.

## What’s the plan?

- [x] Rearrange deployment routes
- [x] Get tests back to green
- [x] Drop legacy redirect routes

## What does it look like?

https://user-images.githubusercontent.com/34030/152976706-b6266095-0858-4fd9-a1b4-3fb6f277b15c.mp4

## How do I test it?

1. `git checkout ui/deployment-routes-refactor`
2. `cd ui && yarn start`
3. [Visit http://localhost:4200](http://localhost:4200)
4. Browse around between deployments
5. Verify that everything works as expected
6. Try recreating #2918 
7. Verify you no longer see the error
